### DISCUSSION TITLE
Show base branch in pull requests list

### DISCUSSION THREAD

**Discussion by driesvints** [community/community#28858] [Pull Requests]
Right now, at @laravel, we add the base branch name to every PR so it's obvious to which branch a PR is targeting when viewing a list of PR's. I'd very much would like to see this natively implemented in GitHub so we don't have to do this anymore. The idea would also be for it to be shown in the closed PR list so you have a historical quick overview of PR's and to which (maybe deleted branch) they were sent to.

See the list here: https://github.com/laravel/framework/pulls

<img width="975" alt="Screenshot 2022-08-09 at 09 46 38" src="https://user-images.githubusercontent.com/594614/183594411-e7136fb5-60f8-4f28-b50d-a18ebf2f2c73.png">

What would be cool is if something like the following would be used:

<img width="748" alt="Screenshot 2022-08-09 at 09 50 14" src="https://user-images.githubusercontent.com/594614/183594740-d3da8154-147c-4592-a760-fa099919cc4e.png">

Where the base branch name is part of the item somewhere. This probably needs to be a bit more visually appealing (I'm not a designer 😅).

I'd love for this to be considered ❤️

**Comment by ghost** [comment]
Where the base branch name is part of the item somewhere. This probably needs to be a bit more visually appealing (I'm not a designer 😅).

I'd love for this to be considered.

**Comment by driesvints** [comment]
Maybe something like this? 

<img width="520" alt="Screenshot 2024-03-26 at 10 47 25" src="https://github.com/community/community/assets/594614/8b12351e-31f9-4b40-b8dd-14aefc2837bc">

**Comment by GromNaN** [comment]
If you use milestones, they are correctly displayed in the list; and you can also filter on them. You can use a github action to automatically set the milestone from the target branch.
<img width="1229" alt="image" src="https://github.com/user-attachments/assets/5268acee-0ae6-415f-964c-3c4f248ce5d4" />

**Reply by simonhamp** [reply]
But that forces you to create new things just for visual convenience... you already have the target branch, why not show it?

**Reply by driesvints** [reply]
@GromNaN but that's not the same thing? I need to see the base branch, not some version?

**Reply by driesvints** [reply]
Talking not only about packages but app projects as well that don't have versioned branches.

**Reply by GromNaN** [reply]
Let me expand on my thoughts: I assumed that you wanted to see the base branch in order to quickly identify which version a PR is for. Laravel uses 1 branch per major version, and the target branch is chosen according to whether it's a bug fix, a new feature with or without breaking change (I don't know the exact strategy).
From there, you can say that the target branch corresponds to the target version. What I suggest is that you use the tool provided by GitHub to identify versions: milestones.
A milestone is not necessarily an exact version number, it can be a major version number. You might have milestone 10.x for the 10.x branch, milestonone 11.x for the 11.x branch and so on.

**Reply by driesvints** [reply]
Yeah, I appreciate the thought but just think that's not worth it. We really need GitHub to surface the base branch here.

**Comment by driesvints** [comment]
In the meantime I've also found this going to be of use on app projects, not just packages with versions. Being able to see the target branch in the pull request view would give me a direct insight in which PR's I'd need to check into.

**Reply by asgrim** [reply]
@driesvints I made a Firefox extension to display the target branch - it is now published :) https://addons.mozilla.org/en-GB/firefox/addon/github-prs-display-target-bran/

It doesn't show it on closed PRs yet, but I plan to eventually. Source: https://github.com/asgrim/gh-pr-display-target-branch

**Reply by driesvints** [reply]
@asgrim nice! I'm not on firefox unfortunately so still hoping that GitHub will get this in some day.
